### PR TITLE
build(aio): jsdoc tag changes

### PR DESCRIPTION
**Pull request by petebacondarwin**
Addresses #22135 and #22385

### CODE REVIEW

**Comment by petebacondarwin** [comment]
Here is a good example of the `description` splitting: https://pr22401-4a9a812.ngbuilds.io/api/animations/AnimationBuilder

The rendered docs look like:

screen shot 2018-02-23 at 14 09 50

While the original source code comment looks like:

screen shot 2018-02-23 at 14 10 09

File: aio/src/styles/0-base/_typography.scss [hunk 1]
```diff
@@ -14,16 +14,6 @@ h1 {
...
 
-h1:after {
-  content: "";
-  display: block;
-  height: 1px;
-  width: 40%;
-  margin: 24px 0px 10px;
-  background: $lightgray;
-  clear: both;
-}
```
Comment (gkalpak) [inline aio/src/styles/0-base/_typography.scss]: This change seems totally unrelated to this commit. Maybe move it in a separate one 😇 
Also, there is one more `h1:after` style in `_typography.scss` and one in `_marketing-layout.scss` that are redundant now.

File: aio/tools/transforms/templates/api/includes/what-it-does.html [hunk 2]
```diff
@@ -1,5 +0,0 @@
-{%- if doc.whatItDoes %}
-<div class="what-it-does info-banner">
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/includes/what-it-does.html]: If we don't use `info-banner` any more here, could we remove the corresponding CSS styles?

File: aio/tools/transforms/templates/api/export-base.template.html [hunk 3]
```diff
@@ -1,10 +1,10 @@
 {% extends 'base.template.html' -%}
 
 {% block body %}
-  {% include "includes/what-it-does.html" %}
+  <p class="short-description">{$ doc.shortDescription | marked $}</p>
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/export-base.template.html]: Is it possible to not have `shortDescription`? In that case, would it be better to omit the `<p>` altogether?

File: aio/tools/transforms/templates/api/class.template.html [hunk 4]
```diff
@@ -4,7 +4,7 @@
 {% extends 'base.template.html' -%}
 
 {% block body %}
-  <p>{$ doc.whatItDoes | marked $}</p>
+  <p class="short-description">{$ doc.shortDescription | marked $}</p>
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/class.template.html]: Is it possible to not have `shortDescription`? In that case, would it be better to omit the `<p>` altogether?

File: aio/tools/transforms/angular-api-package/tag-defs/usageNotes.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+module.exports = function(log, createDocMessage) {
```
Comment (gkalpak) [inline aio/tools/transforms/angular-api-package/tag-defs/usageNotes.js]: Unused deps :grin:

File: aio/tools/transforms/angular-api-package/processors/splitDescription.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+module.exports = function splitDescription(log, createDocMessage) {
+  return {
+    $runAfter: ['tags-extracted', 'migrateLegacyJSDocTags'],
+    $runBefore: ['processing-docs'],
+    docTypes: [],
+    $process(docs) {
+      docs.forEach(doc => {
+        if (this.docTypes.indexOf(doc.docType) !== -1 && doc.description !== undefined) {
+          const description = doc.description.trim();
+          const endOfParagraph = description.indexOf('\n\n');
```
Comment (gkalpak) [inline aio/tools/transforms/angular-api-package/processors/splitDescription.js]: Is it possible to have whitespace between the two `\n`?

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/tools/transforms/angular-api-package/processors/splitDescription.js (hunk 6 above)
Comment (petebacondarwin) [reply aio/tools/transforms/angular-api-package/processors/splitDescription.js]: That would be nasty 😠 but OK, I guess we can handle that too.

File: aio/tools/transforms/templates/api/class.template.html (hunk 4 above)
Comment (petebacondarwin) [reply aio/tools/transforms/templates/api/class.template.html]: I think that the short description will come under the "required" parts of the API docs.

File: aio/tools/transforms/templates/api/export-base.template.html (hunk 3 above)
Comment (petebacondarwin) [reply aio/tools/transforms/templates/api/export-base.template.html]: https://github.com/angular/angular/pull/22401#discussion_r170758306

File: aio/tools/transforms/angular-api-package/processors/splitDescription.spec.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
+const testPackage = require('../../helpers/test-package');
+const processorFactory = require('./splitDescription');
+const Dgeni = require('dgeni');
+
+fdescribe('splitDescription processor', () => {
```
Comment (petebacondarwin) [inline aio/tools/transforms/angular-api-package/processors/splitDescription.spec.js]: Oops

File: aio/src/styles/0-base/_typography.scss [hunk 8]
```diff
@@ -60,13 +50,7 @@ h6 {
   clear: both;
 }
 
-h1 {
-  @media screen and (max-width: 600px) {
-    margin: 0;
```
Comment (gkalpak) [inline aio/src/styles/0-base/_typography.scss]: Why remove this style?

**Review by gkalpak — APPROVED** [review]
:100:

File: aio/src/styles/0-base/_typography.scss (hunk 8 above)
Comment (petebacondarwin) [reply aio/src/styles/0-base/_typography.scss]: Previously there was an underline beneath `h1` tags. This underline was given the appropriate margin (`8px 0`) for the surrounding text.  Now that the underline has gone the margins are set by the following rule.

That being said, I now see that this rule also removes margin from before the `h1` on narrow screens so I will put that back in.

Comment (petebacondarwin) [reply aio/src/styles/0-base/_typography.scss]: See last fixup commit 1443bfb190856ebd60ddbbaaf987a2ce3032be18

**Pull request closed without merging**